dataRetriever/errors.go

```diff
@@ -184,3 +184,6 @@ var ErrEmptyHeaderRequestTopic = errors.New("empty header request topic")
 
 // ErrInvalidMaxTxRequest signals that max tx request is too small
 var ErrInvalidMaxTxRequest = errors.New("max tx request number is invalid")
+
+// ErrNilPeersListCreator signals that a nil peers list creator implementation has been provided
+var ErrNilPeersListCreator = errors.New("nil peers list creator provided")
```

Comment:
Can change everywhere to `PeerList` (peer - singular) since List implies multiple items